### ISSUE TITLE
Transaction error in isolate with batch (computeWithDatabase)

### ISSUE THREAD

**Issue by austinried** [simolus3/moor#2344]
I'm trying to set up and use `computeWithDatabase` for the first time with a batch, I believe I've got it set it right because the batch I'm running works the first time, but the second time I try to use it I get this error:

```
E/flutter ( 4926): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument (transactionId): Does not reference a transaction. This might happen if you don't await all operations made inside a transaction, in which case the transaction might complete with pending operations.: null
E/flutter ( 4926): #0      DriftCommunication.request
communication.dart:116
E/flutter ( 4926): #1      _RemoteTransactionExecutor._sendAction
client_impl.dart:220
E/flutter ( 4926): #2      _RemoteTransactionExecutor.rollback
client_impl.dart:228
E/flutter ( 4926): #3      Batch._commit
batch.dart:203
E/flutter ( 4926): <asynchronous suspension>
E/flutter ( 4926): #4      SyncService._saveAlbum.<anonymous closure>
sync_service.dart:112
E/flutter ( 4926): <asynchronous suspension>
E/flutter ( 4926): #5      ComputeWithDriftIsolate.computeWithDatabase.<anonymous closure>
isolate.dart:316
E/flutter ( 4926): <asynchronous suspension>
E/flutter ( 4926): #6      _RemoteRunner._run (dart:isolate:956:18)
E/flutter ( 4926): <asynchronous suspension>
```

The batch works fine outside of an isolate, I'd just rather move it into one as it seems to be causing some frame drops on the UI thread. Here's the function I'm using:

```dart
  Future<void> _saveAlbum(Album album) {
    return db.computeWithDatabase(
      connect: SubtracksDatabase.connection,
      computation: (db) async {
        await db.batch((batch) {
          batch.insertAllOnConflictUpdate(db.albums, [album.toInsertable()]);
          batch.insertAllOnConflictUpdate(
            db.songs,
            album.songs.map((e) => e.toInsertable()),
          );

          final songIds = {for (var a in album.songs) a.id};
          batch.deleteWhere(
            db.songs,
            (tbl) =>
                tbl.sourceId.equals(album.sourceId) &
                tbl.albumId.equals(album.id) &
                tbl.id.isNotIn(songIds),
          );
        });
      },
    );
  }
```

**Comment by simolus3** [comment]
Thanks for the report. This looks like a bug in drift, but I was unable to reproduce it locally (with a similar batch in `computeWithDatabase`). Do you think there's a way for you to prepare a standalone example project that reproduces this error if it's not too much work?

**Comment by austinried** [comment]
This is from a decent sized project so I imagine there could be something interfering here, but I could try.

Also just want to make a small correction, it actually doesn't work the first time like I thought, it just appears to work in that I see the whole transaction from the batch show up in the logs, including `Drift: Sent COMMIT TRANSACTION with args []` at the end, but it appears to never actually commit.

**Comment by austinried** [comment]
Actually that was easier than I thought, I just ripped the smallest parts I could out of my app and shoved it into the default flutter android app. Repo is here: https://github.com/austinried/drift_batch

**Comment by rod-moraes** [comment]
@austinried I was having problems with batch when I updated the Flutter version to 3.7 and when I go back to 3.3.8 everything works... could this be related to your problem as well?

**Comment by simolus3** [comment]
Thanks! I can reproduce the problem with that example, but it seems like (for that example), the problem goes away when using the upcoming drift version.

If you add this to your pubspec and restart, does the error still happen in your larger project?

```yaml
dependency_overrides:
  drift:
    git:
      url: https://github.com/simolus3/drift.git
      ref: develop
      path: drift
```

**Comment by austinried** [comment]
Looks like that works in my real project too, thanks! I'll just stick with that override for now then.

**Comment by simolus3** [comment]
Should be fixed in `drift` version `1.6.0`.
